sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2251,6 +2251,22 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val STATE_STORE_PARTITION_METRICS_REPORT_LIMIT =
+    buildConf("spark.sql.streaming.stateStore.numPartitionMetricsToReport")
+      .internal()
+      .doc(
+        "Maximum number of partition-level metrics to include in state store progress " +
+          "reporting. The default limit is 20% of the number of cores (with a minimum of 1 " +
+          "partition) and with a cap of 10. This limits the metrics to the N partitions with " +
+          "the smallest values to prevent the progress report from becoming too large."
+      )
+      .version("4.0.0")
+      .intConf
+      .checkValue(k => k >= 0, "Must be greater than or equal to 0")
+      .createWithDefault(
+        Math.min(10, Math.min(1, SHUFFLE_PARTITIONS.defaultValue.getOrElse(200) / 5))
```

Comment:
Not sure we need to getOrElse with 200 here. Shuffle partition is INT here and always set > 0. If someone updates the default of shuffle partition, they will likely not update this.